### PR TITLE
ENH: add fill argument to AnchoredSizeBar

### PR DESCRIPTION
**Pull request by magnunor**
## PR Summary

This pull request introduces a optional AnchoredSizeBar argument for filling the rectangle used in the sizebar.

Having the option to have a filled rectangle is useful when plotting scalebars for microscopy images, where this is the conventional way.

```Python
import matplotlib.pyplot as plt
from mpl_toolkits.axes_grid1.anchored_artists import AnchoredSizeBar

fig, ax = plt.subplots(figsize=(3, 3))

bar0 = AnchoredSizeBar(ax.transData, 0.3, '0.3', loc=3, frameon=False, size_vertical=0.05)
ax.add_artist(bar0)
bar1 = AnchoredSizeBar(ax.transData, 0.3, '0.3', loc=4, frameon=False, fill=True, size_vertical=0.05)
ax.add_artist(bar1)
fig.savefig("sizebar_fill.png")
```
sizebar_fill

## PR Checklist

- [ ] Has Pytest style unit tests
- [x] Code is PEP 8 compliant 
- [ ] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant

### CODE REVIEW

File: lib/mpl_toolkits/axes_grid1/anchored_artists.py [hunk 1]
```diff
@@ -232,7 +232,7 @@ class AnchoredSizeBar(AnchoredOffsetbox):
     def __init__(self, transform, size, label, loc,
                  pad=0.1, borderpad=0.1, sep=2,
                  frameon=True, size_vertical=0, color='black',
-                 label_top=False, fontproperties=None,
+                 fill=False, label_top=False, fontproperties=None,
```
Comment (tacaswell) [inline lib/mpl_toolkits/axes_grid1/anchored_artists.py]: please put this last to not break the API.  Technically someone could be passing in 12 positional arguments.

**Comment by tacaswell** [comment]
In general :+1: but needs a few tweaks.

**Comment by magnunor** [comment]
Fixed!

I wasn't completely sure how to format the whats_new, since there were some variations in the already existing files in that folder.

**Comment by tacaswell** [comment]
This was last changed in e5b5425d8eb2242912bdc977bc1d945a3fb6a320  to fix #5757 .  This change should not affect that (as the default does not change).  It looks like what is going on is:

 - filled polygons do not get and edge by default now
 - patches are filled by default
 - the default height of the scale bar is 0 so in the default case, fill needs to be `False` for it to show up (that is #5757)
 - in this case with finite height, the fill should be `True`

It would not be unreasonable to change the default value of fill based on if the height is >0

```python
if fill is None:
    fill = size_vertical > 0
```

Bit trickier to document, but probably more user-friendly in the long run.

If we were to do this again, I would say that the default height should be `0.01` or something like that, but :man_shrugging: .

**Comment by magnunor** [comment]
Functionality being implemented in https://github.com/matplotlib/matplotlib/pull/8835

**Pull request closed without merging**